### PR TITLE
add frame/handler support for ping req, ping res, claim, and cancel.

### PR DESCRIPTION
**Pull request by ShanniLi**
r @jcorbin @kriskowal 

support operation claim
claim frame parser
internal handler support

support operation cancel
cancel frame parser
internal handler support

ping req/res frame parsers
internal handler support

### CODE REVIEW

**Comment by kriskowal** [comment]
This looks good to me. When this lands, check off relevant parts on https://github.com/uber/tchannel/issues/122

**Comment by kriskowal** [comment]
And wait for @jcorbin to weigh in.

File: node/v2/handler.js [hunk 1]
```diff
@@ -50,8 +50,12 @@ function TChannelV2Handler(options) {
     self.callIncomingErrorEvent = self.defineEvent('callIncomingError');
     self.callIncomingRequestEvent = self.defineEvent('callIncomingRequest');
     self.callIncomingResponseEvent = self.defineEvent('callIncomingResponse');
+    self.cancelEvent = self.defineEvent('cancel');
     self.initRequestEvent = self.defineEvent('initRequest');
     self.initResponseEvent = self.defineEvent('initResponse');
+    self.claimEvent = self.defineEvent('claim');
```
Comment (jcorbin) [inline node/v2/handler.js]: let's move this up to b/w L53/54 to maintain alpahabetic ordering

File: node/v2/claim.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+var Tracing = require('./tracing');
+
+// tracing:25
+function Claim(tracing) {
+    var self = this;
+    self.type = Claim.TypeCode;
+    self.tracing = tracing || Tracing.emptyTracing;
+}
+
+Claim.TypeCode = 0xC1;
```
Comment (jcorbin) [inline node/v2/claim.js]: The convention in the node implementation so far is to lowercase hex literals, would be nice to maintain that here and in Cancel.TypeCode above

File: node/v2/handler.js [hunk 3]
```diff
@@ -208,6 +220,12 @@ TChannelV2Handler.prototype.callResponseFrameHandled = function callResponseFram
     callback();
 };
 
+TChannelV2Handler.prototype.handleCancel = function handleCancel(frame, callback) {
```
Comment (jcorbin) [inline node/v2/handler.js]: Let's at add a TODO noting that we should implement clearing of `self.streaming{Req,Res}`.

- should clear out of self.streaming{Req,Res}
- but they're currently keyed by id not tracing
- however we probably can't just shift the primary key of these collections
- rather we'll probably need to add a secondary index to protect against pathological "zero tracing" being sent... unless of course we make that a hard error

Similar concerns apply at the `TChannelConnection.requests.{in,out} layer`.

**Comment by jcorbin** [comment]
:+1: optional feedback, this is good enough to merge and go forward

File: node/v2/claim.js (hunk 2 above)
Comment (ShanniLi) [inline node/v2/claim.js]: fixed.

File: node/v2/handler.js (hunk 3 above)
Comment (ShanniLi) [inline node/v2/handler.js]: TODO added.

**Pull request merged**